irteus/irtmodel.l

```diff
@@ -2396,10 +2396,18 @@
        )
      pairs))
   (:self-collision-check
-   (&key (mode :all) (pairs (send self :collision-check-pairs)) (collision-func 'pqp-collision-check))
+   (&key (mode :all) (pairs (send self :collision-check-pairs)) (collision-func 'pqp-collision-check) (distance-func 'pqp-collision-distance) (fat nil))
+   "calculate self collision chaeck
+      :mode (:all or :first) ; returns first collided link pair of all collided link pair
+      :pairs (list (cons <link1> <link2>) ....) ; linke pair to be checked
+      :fat (number) if number is set, any link pair that has smaller distance than this fat threshold will regard as collided"
    (let ((cpairs) (col-count 0))
      (dolist (p pairs)
-       (let ((colp (/= (funcall collision-func (car p) (cdr p)) 0)))
+       (let ((colp
```

Comment:
Thanks, I understand the motivation of this PR.
For this purpose, it would be better to another name for fat as you said.,
e.g., `fat` => `min-distance`, because fatting collision shape is not theoretically same as changing distance.